### PR TITLE
fix: relation schemas should not inherit `title` from the source model's schema

### PR DESCRIPTION
**Pull request by jannyHou**
Fix https://github.com/strongloop/loopback-next/issues/5088

When building the relation schema, the `title` field should be omitted.
See details in comment https://github.com/strongloop/loopback-next/issues/5088#issuecomment-613069334

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/repository-json-schema/src/build-schema.ts [hunk 1]
```diff
@@ -501,7 +502,11 @@ export function modelToJsonSchema<T extends object>(
       result.properties = result.properties ?? {};
       const relMeta = meta.relations[r];
       const targetType = resolveType(relMeta.target);
-      const targetSchema = getJsonSchema(targetType, options);
+      // omit the title for related schema
+      const targetSchema = getJsonSchema(
+        targetType,
+        _.omit(options, ['title']),
```
Comment (raymondfeng) [inline packages/repository-json-schema/src/build-schema.ts:508]: The `omit` function can be implemented as:

```ts
function omit(options: object, ...props: string[]} {
  const copy = {...options};
  for (const p of props) {
    delete copy[p];
  }
  return copy;
}

Then just call `omit(options, 'title')`

Comment (raymondfeng) [reply packages/repository-json-schema/src/build-schema.ts:508]: Let's not add `lodash` as a dependency unless necessary.

Comment (bajtos) [reply packages/repository-json-schema/src/build-schema.ts:508]: +1 to avoiding lodash dependency.

I think this line can be also rewritten as follows:

```suggestion
        {...options, title: undefined},
```

File: packages/repository-json-schema/src/__tests__/unit/build-schema.unit.ts [hunk 2]
```diff
@@ -246,6 +248,56 @@ describe('build-schema', () => {
...
+        color: {type: 'string'},
+      });
+      // No circular references in definitions
+      expect(schema.definitions).to.containEql({
+        ChildWithRelations: {
+          title: 'ChildWithRelations',
+          description: '(Schema options: { includeRelations: true })',
+          properties: {name: {type: 'string'}},
+          additionalProperties: false,
+        },
```
Comment (agnes512) [inline packages/repository-json-schema/src/__tests__/unit/build-schema.unit.ts:297]: LGTM. Just have a question. If `Parent` has many `Parent`, would the title here be `PatentWIthRelations`?

Comment (jannyHou) [reply packages/repository-json-schema/src/__tests__/unit/build-schema.unit.ts:297]: @agnes512 For self relations, the same schema is cached so that [line 419](https://github.com/strongloop/loopback-next/blob/344784be9d9b7624952db1f17473d5308ba2ae70/packages/repository-json-schema/src/build-schema.ts#L419) would return.

We also have a test case for it just above the new ones I added in this PR :)

**Comment by jannyHou** [comment]
@raymondfeng @bajtos @agnes512 thank you for review, removed lodash 👍 

> I find the PR title misleading.
True. I changed it to "omit title in options when generating relation schemas"

File: packages/repository-json-schema/src/build-schema.ts [hunk 3]
```diff
@@ -501,7 +501,14 @@ export function modelToJsonSchema<T extends object>(
       result.properties = result.properties ?? {};
       const relMeta = meta.relations[r];
       const targetType = resolveType(relMeta.target);
-      const targetSchema = getJsonSchema(targetType, options);
+
+      const targetOptions = {...options};
+      if (targetOptions.title) {
+        // Omit the title for related schema
```
Comment (bajtos) [inline packages/repository-json-schema/src/build-schema.ts:507]: Perhaps add a code comment to explain why is this needed?

**Comment by jannyHou** [comment]
> The idea is to describe the effect observed by LB users, not the implementation details (those should be explained in the commit/PR details).

Agreed 👍 renamed it to "relation schemas should not inherit `title` from the source model's schema", and also added comments to the code+tests.
